v-next/hardhat-network-helpers/test/helpers/mocked-rpc-method.ts

```diff
@@ -0,0 +1,41 @@
+import type {
+  HookContext,
+  NetworkHooks,
+} from "@ignored/hardhat-vnext/types/hooks";
+import type {
+  ChainType,
+  NetworkConnection,
+} from "@ignored/hardhat-vnext/types/network";
+import type {
+  JsonRpcRequest,
+  JsonRpcResponse,
+} from "@ignored/hardhat-vnext/types/providers";
+
+// This network hook handler will returned a mocked value when the method "web3_clientVersion" is called.
+// This will simulate a non-test network.
+export default async (): Promise<Partial<NetworkHooks>> => {
+  const handlers: Partial<NetworkHooks> = {
+    onRequest: async <ChainTypeT extends ChainType | string>(
+      context: HookContext,
+      networkConnection: NetworkConnection<ChainTypeT>,
+      jsonRpcRequest: JsonRpcRequest,
+      next: (
+        nextContext: HookContext,
+        nextNetworkConnection: NetworkConnection<ChainTypeT>,
+        nextJsonRpcRequest: JsonRpcRequest,
+      ) => Promise<JsonRpcResponse>,
+    ) => {
+      if (jsonRpcRequest.method === "web3_clientVersion") {
+        return {
+          jsonrpc: "2.0",
+          id: jsonRpcRequest.id,
+          result: "non-test-network",
```

Comment:
Do you think it's worth testing the scenarios where we use other development network names that we support? If so, we could make this value configurable here and reuse that mock for the happy cases as well.

Reply:
It would complicate the testing a bit, because I will have to use a different logic to pass the mocked value. I think this is a good compromise between testability and coverage

Reply:
Ah, now I see how we pass this mock through the config. Alright, that makes sense.